tests/ccgbugs/m1/defs.nim

```diff
@@ -0,0 +1,2 @@
+type MyObj* = object
+  field1*: int
```

Comment:
an additional test here would be needed, constructed the following way:
* same name
* same field names
* different field order

such an object, before this commit, will be compiled correctly but its data would be written according to the wrong object type in certain cases leading to runtime corruption